compiler/src/dotty/tools/dotc/typer/Implicits.scala

```diff
@@ -1004,7 +1005,7 @@ trait Implicits { self: Typer =>
         val generated2 =
           if (cand.isExtension) Applications.ExtMethodApply(generated1).withType(generated1.tpe)
           else generated1
-        SearchSuccess(generated2, ref, cand.level)(ctx.typerState)
+        SearchSuccess(generated2, ref, cand.level)(ctx.typerState, ctx.gadt)
```

Comment:
Would it make sense to combine typerState and GADT map?

Reply:
I'm keeping my eye on it. I tried to do that, but couldn't find a clean approach. The problem is we don't want to set fresh typer state when we set fresh GADT state, and vice versa - it's really easiest if the two are separate objects. If current approach turns out to be too clumsy, I'll find a way to merge the two.